tests/integration/test_examples.py

```diff
@@ -119,3 +120,33 @@ def test_run_caching(examples_dir: Path):
     assert second_run.steps[1].is_cached
     assert not second_run.steps[2].is_cached
     assert not second_run.steps[3].is_cached
+
+
+def test_run_mlflow(examples_dir: Path):
+    """Testing the functionality of the quickstart example
+
+    Args:
+        Temporary folder containing all examples including the run_examples
+        bash script.
+    """
+    local_example = LocalExample(examples_dir / MLFLOW, name=MLFLOW)
+
+    bash_script_location = examples_dir / EXAMPLES_RUN_SCRIPT
+    local_example.run_example(bash_file=str(bash_script_location), force=True)
+
+    # Verify the example run was successful
+    repo = Repository(path=str(local_example.path))
+    pipeline = repo.get_pipelines()[0]
+    assert pipeline.name == "mlflow_example_pipeline"
+
+    first_run = pipeline.runs[-2]
+    second_run = pipeline.runs[-1]
+
+    # Both runs should be completed
+    assert first_run.status == ExecutionStatus.COMPLETED
+    assert second_run.status == ExecutionStatus.COMPLETED
+
+    for step in first_run.steps:
+        assert step.status == ExecutionStatus.COMPLETED
+    for step in second_run.steps:
+        assert step.status == ExecutionStatus.COMPLETED
```

Comment:
More of a general question: should we include some integration specific validation steps in these example tests ? For example, in this particular case, checking that MLFlow artifacts and/or metrics are indeed saved using the right experiment name in the local mlflow backend ?

Reply:
I agree 100%. However there are some issues regarding the integration tests and how the repo ist set within the subprocess calls that made it a bit difficult. I should have added a TODO to have a more in depth look at this.